opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -2272,52 +2253,91 @@ void CpGrid::addLgrsUpdateLeafView(const std::vector<std::array<int,3>>& cells_p
-
+            
             // Global ids for cells (for owned cells)
             for (const auto& element : elements(levelGridView(level))) {
                 if (element.partitionType() == InteriorEntity) {
                     localToGlobal_cells_per_level[level - 1][element.index()] = min_globalId_cell_in_proc;
                     ++min_globalId_cell_in_proc;
                 }
             }
-            // Global ids for points (global ids might be overestimated, but still unique).
             for (const auto& point : vertices(levelGridView(level))) {
-                // If point coincides with an existing corner from level zero, then it does not need a new global id.
-                if ( !(*current_data_)[level]->corner_history_.empty() ) {
-                    const auto& bornLevel_bornIdx =  (*current_data_)[level]->corner_history_[point.index()];
-                    if (bornLevel_bornIdx[0] != -1)  { // Corner in the refined grid coincides with a corner from level 0.
-                        // Therefore, search and assign the global id of the previous existing equivalent corner.
-                        const auto& equivPoint = cpgrid::Entity<3>(*( (*current_data_)[bornLevel_bornIdx[0]]), bornLevel_bornIdx[1], true);
-                        localToGlobal_points_per_level[level-1][point.index()] =
-                            current_data_->front()->global_id_set_->id( equivPoint );
-                    }
-                    else {
-                        // Assign a global ID to points that do not coincide with any corners from level zero.
-                        // This ID may be overwritten later when parallel indices and level cell index sets
-                        // are defined for all level grids, particularly in cases where LGRs may be distributed
-                        // across processes.
-                        localToGlobal_points_per_level[level-1][point.index()] = min_globalId_point_in_proc;
-                        ++min_globalId_point_in_proc;
-                    }
+                // Check if it coincides with a corner from level zero. In that case, no global id is needed.
+                const auto& bornLevel_bornIdx =  (*current_data_)[level]->corner_history_[point.index()];
+                if (bornLevel_bornIdx[0] != -1)  { // Corner in the refined grid coincides with a corner from level 0.
+                    // Therefore, search and assign the global id of the previous existing equivalent corner.
+                    const auto& equivPoint = cpgrid::Entity<3>(*( (*current_data_)[bornLevel_bornIdx[0]]), bornLevel_bornIdx[1], true);
+                    localToGlobal_points_per_level[level-1][point.index()] = current_data_->front()->global_id_set_->id( equivPoint );
+                }
+                else {
+                    // Assign new global id only to non-overlap points that do not coincide with
```

Comment:
Indeed, the comment is wrong. At this stage in the code, using Dune::Partitions::interior in vertices(..., ...::interior) actually includes all points, since the partition type for points in the refined-level grids hasn’t been set yet. This is handled a bit further down, after cellIndexSet is resized and the partition types for cells in the refined grids are computed.

Partition type of points is a bit tricky since there are 4 types: interior, border, front, overlap. This classification requires that both cell and face partition types are already defined.

Thanks for catching that! I’ll remove the incorrect comment.

Reply:
Thanks for the clarification. Should we add a comment about partitions not being ready to use, if it is not there already? This might not be obvious to everyone.

Reply:
There is a  comment [here](https://github.com/OPM/opm-grid/blob/abe6a69546031cc4a4dcf0c02c2290c216c452e4/opm/grid/cpgrid/CpGrid.cpp#L2200) but it's not 100% clear. I'll re-write that, thanks!